### PR TITLE
[TKET-1567] Add aas in rv3.1

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

**Comment by cqc-melf** [comment]
This is my suggestion to add aas routing and labelling into rv3.1.

**Comment by cqc-melf** [comment]
Take a look on https://github.com/CQCL/tket/pull/206 for the old PR.

File: pytket/tests/circuit_test.py [hunk 1]
```diff
@@ -690,7 +690,7 @@ def test_phase_polybox() -> None:
     n_qb = 1
     qubit_indices = {Qubit(0): 1}
     phase_polynomial = {(True,): 0.1, (False,): 0.1}
-    linear_transformation = np.array([[1, 0], [0, 1]])
+    linear_transformation = np.array([[1, 0], [0, 1]])  # TODO, this is wrong
```
Comment (sjdilkes) [inline pytket/tests/circuit_test.py:693]: Check TODO?

File: pytket/tests/mapping_test.py [hunk 2]
```diff
@@ -119,6 +119,127 @@ def test_LexiRouteRoutingMethod() -> None:
...
+    test_c.add_phasepolybox(p_box, [0, 1, 2])
+    test_c.CX(0, 1)
+
+    nodes = [Node("test", 0), Node("test", 1), Node("test", 2)]
+    test_a = Architecture([[nodes[0], nodes[1]], [nodes[1], nodes[2]]])
+    test_mm = MappingManager(test_a)
+    test_mm.route_circuit(
+        test_c,
+        [
+            AASRouteRoutingMethod(1),
```
Comment (sjdilkes) [inline pytket/tests/mapping_test.py:224]: I'm finding this pretty satisfying!!

**Review by sjdilkes — CHANGES_REQUESTED** [review]
Just the todo!

**Review by sjdilkes — DISMISSED** [review]

File: tket/src/Mapping/AASLabelling.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "Mapping/AASLabelling.hpp"
+
+namespace tket {
+
+/*bool AASLabellingMethod::check_method(
```
Comment (sjdilkes) [inline tket/src/Mapping/AASLabelling.cpp:19]: remove

File: tket/src/Mapping/AASRoute.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+#include "Mapping/AASRoute.hpp"
+
+namespace tket {
+
+AASRouteRoutingMethod::AASRouteRoutingMethod(
+    unsigned _aaslookahead, aas::CNotSynthType _cnotsynthtype)
+    : cnotsynthtype_(_cnotsynthtype), aaslookahead_(_aaslookahead) {}
+
+/*bool AASRouteRoutingMethod::check_method(
```
Comment (sjdilkes) [inline tket/src/Mapping/AASRoute.cpp:23]: remove

**Review by sjdilkes — CHANGES_REQUESTED** [review]
Just a few things

File: .github/workflows/build_and_test.yml [hunk 5]
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - main
       - develop
+      - feature/RV3.1
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply .github/workflows/build_and_test.yml]: done

File: tket/src/Mapping/AASLabelling.cpp (hunk 3 above)
Comment (cqc-melf) [reply tket/src/Mapping/AASLabelling.cpp]: done

**Review by sjdilkes — APPROVED** [review]
Nice

**Pull request merged**